src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1420,16 +1430,83 @@ private IEnumerable<TypeInferenceInfo> InferTypeInImplicitArrayCreation(Implicit
                 return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInPropertyPattern(
+                PropertyPatternSyntax propertyPattern,
+                SyntaxNode child)
+            {
+                // something like: { ... sub patterns ... } or like: Blah { ... sub patterns ... }
+                // in the latter case, we know the type, because it's explicitly stated.  In the former
+                // we have to walk higher to figure it out.
+                if (propertyPattern.Type != null)
+                {
+                    return GetTypes(propertyPattern.Type);
+                }
+
+                return InferTypes(propertyPattern);
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInPropertySubpattern(
+                PropertySubpatternSyntax propertySubpattern,
+                SyntaxNode child)
+            {
+                return InferTypes(propertySubpattern);
+            }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInSubpatternElement(
+                SubpatternElementSyntax subpatternElement,
+                SyntaxNode child)
+            {
+                // we have  { X: ... }.  The type of ... is whatever the type of 'X' is in its
+                // parent type.  So look up the parent type first, then find the X member in it
+                // and use that type.
+                if (child == subpatternElement.Pattern &&
+                    subpatternElement.NameColon.Name is IdentifierNameSyntax identifier)
+                {
+                    var parentTypes = this.InferTypes(subpatternElement);
+                    var memberTypes = 
+                        parentTypes.SelectMany(ti => GetMemberReturnTypes(ti.InferredType, identifier.Identifier.ValueText))
+                                   .SelectAsArray(t => new TypeInferenceInfo(t));
+
```

Comment:
why is all this logic necessary? doesn't the compiler return a symbol for X?

Reply:
Good question.  Let me try that.

Reply:
The semantic model is not implemented for this syntactic location yet.

Reply:
@gafter Is the expectation that it *will* be at some later point?  I just want to know what i should expect to work in the future so i can write the feature code in the appropriate manner :)  

For example, you explained in the past how patterns don't have types, so i know to not even expect to be able to use .GetTypeInfo on a pattern.  But the rules about things like getting symbol info on these parts of a pattern are a bit less clear.

Thanks!

Reply:
Yes, I intend that identifiers referenced within a pattern will have SymbolInfo working. I will work on that next (though I am out for about a week, so it will be more than a week before I have it working).

Reply:
Sounds good.  I'll wait for that to be done.